### PR TITLE
fix/api/topics

### LINKED ISSUE #151: API: Fix /api/topics
Ensure that the endpoint works by testing it.

### PR DESCRIPTION
**Pull request by MaherHussain**
# Description

 this PR is about fixing topics endpoint and test it on swagger .

Fixes #151

# How to test?

tested on swagger 
https://ibb.co/zXZRR2s

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by Yaser-Aldammad — DISMISSED** [review]
good

File: src/server/api/controllers/topics.controller.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const getTopics = async () => {
 const getTopicById = async (id) => {
   try {
     const topics = await knex('topics')
-      .select('topics.id as id', 'title')
+      .select('topics.id as id', 'topic_name')
       .where({ id });
     if (topics.length === 0) {
```
Comment (Elia-Git) [inline src/server/api/controllers/topics.controller.js:17]: In line nr 17, typo error? , in this  if(topics.length===0) statement "topics" should be "topic" instead.

File: src/server/api/controllers/topics.controller.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const Error = require('../lib/utils/http-error');
 
 const getTopics = async () => {
   try {
-    return await knex('topics').select('topics.id', 'topics.title');
+    return await knex('topics').select('topics.id', 'topics.topic_name');
```
Comment (tpetrina) [inline src/server/api/controllers/topics.controller.js:6]: Please adjust to return as `topicName`.

File: src/server/api/controllers/topics.controller.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ const getTopics = async () => {
 const getTopicById = async (id) => {
   try {
     const topics = await knex('topics')
-      .select('topics.id as id', 'title')
+      .select('topics.id as id', 'topic_name')
```
Comment (tpetrina) [inline src/server/api/controllers/topics.controller.js:15]: Please adjust to return as `topicName`.

**Review by tpetrina — CHANGES_REQUESTED** [review]

File: src/server/api/controllers/topics.controller.js (hunk 1 above)
Comment (tpetrina) [reply src/server/api/controllers/topics.controller.js:20]: You always get a collection back - and then you check how many there are inside.

**Review by tpetrina — APPROVED** [review]
Excellent, ship it!

**Review by Yaser-Aldammad — APPROVED** [review]
well done

**Pull request merged**